google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageReadChannel.java

```diff
@@ -41,27 +42,42 @@
 @ThreadSafe
 final class CloudStorageReadChannel implements SeekableByteChannel {
 
-  private final ReadChannel channel;
+  private final Storage gcsStorage;
+  private final BlobId file;
+  // max # of times we may reopen the file
+  private final int maxReopen;
+  // how many times we re-opened the file
+  private int reopens;
+  private ReadChannel channel;
   private long position;
   private long size;
 
+  /**
+   * @param maxReopen max. number of times to try re-opening the channel if it closes on us unexpectedly.
+   */
   @CheckReturnValue
   @SuppressWarnings("resource")
-  static CloudStorageReadChannel create(Storage gcsStorage, BlobId file, long position)
+  static CloudStorageReadChannel create(Storage gcsStorage, BlobId file, long position, int maxReopen)
```

Comment:
Be consistent with the other place that the max reopen concept is used (my recommendation: `maxChannelReopens`.)